libraries/AP_NavEKF3/AP_NavEKF3_OptFlowFusion.cpp

```diff
@@ -112,7 +112,7 @@ void NavEKF3_core::EstimateTerrainOffset()
 
         // in addition to a terrain gradient error model, we also have the growth in uncertainty due to the copters vertical velocity
         float timeLapsed = MIN(0.001f * (imuSampleTime_ms - timeAtLastAuxEKF_ms), 1.0f);
-        float Pincrement = (distanceTravelledSq * sq(0.01f*float(frontend->gndGradientSigma))) + sq(timeLapsed)*P[6][6];
+        float Pincrement = (distanceTravelledSq * sq(frontend->_terrGradMax)) + sq(timeLapsed)*P[5][5];
```

Comment:
I have to ask if the change at the end of line (P matrix) was on purpose or a copy error.

Reply:
The vertical velocity state for EKF2 is at index 5 and for EKF3 it is at Index 6, so given this is EKF3, then yes good catch, that was a copy error.

Reply:
See 
https://github.com/ArduPilot/ardupilot/pull/10539/commits/031a946ec5cc152f7eec5b9cd55bb22e329e87b0